vortex-datetime-dtype/src/unit.rs

```diff
@@ -26,6 +27,18 @@ pub enum TimeUnit {
     D,
 }
 
+impl TimeUnit {
+    pub fn to_jiff_span(&self, v: i64) -> Result<Span, jiff::Error> {
```

Comment:
this should probably return `VortexResult<Span>`, and just let the jiff Error convert into VortexResult